### ISSUE TITLE
Does not build in CMake Release mode

### ISSUE THREAD

**Issue by voyager1998** [UM-ARM-Lab/arc_utilities#74]
When using `-Werror=unused-variable` and `-DCMAKE_BUILD_TYPE=Release`, `assert` statements do not count as a variable usage. Code snippets like the following will fail:

```
const size_t bytes_used = arc_utilities::SerializeEigen(vec, buffer);
const auto deserialized = arc_utilities::DeserializeEigen<EigenType>(buffer, 0);
assert(deserialized.second == bytes_used);
```

**Comment by dmcconachie** [comment]
This repo is not currently intended to be used with `CMAKE_BUILD_TYPE`. This is something that probably ought to be supported but currently isn't. See https://github.com/UM-ARM-Lab/arc_utilities/issues/56.

**Comment by bsaund** [comment]
https://github.com/UM-ARM-Lab/arc_utilities/pull/78
https://github.com/UM-ARM-Lab/arc_utilities/pull/76
TravisCI is currently building in Release and Debug

**Comment by a-price** [comment]
Addressed by #78 asserts.
